pkg/ccl/backupccl/restore_job.go

```diff
@@ -448,6 +463,36 @@ func restore(
 		tasks = append(tasks, generativeCheckpointLoop)
 	}
 
+	procCompleteCh := make(chan struct{})
+	// countCompletedProcLoop is responsible for counting the number of completed
+	// processors. The goroutine returns a retryable error such that the job can
+	// replan if there are too many lagging nodes.
+	countCompletedProcLoop := func(ctx context.Context) error {
+		var timer timeutil.Timer
+		defer timer.Stop()
+		done := ctx.Done()
+		timer.Reset(replanFrequency.Get(&execCtx.ExecCfg().Settings.SV))
```

Comment:
Note that you could skip this reset and then not start the timer until the very first processor ends.